AndroidHomeApplication/app/src/main/java/com/example/androidhomeapplication/ui/moviesList/FragmentMoviesList.kt

```diff
@@ -32,25 +36,26 @@ class FragmentMoviesList : Fragment(R.layout.fragment_movies_list) {
         super.onViewCreated(view, savedInstanceState)
 
         binding.cinemaRv.layoutManager = GridLayoutManager(context, 2)
-        binding.cinemaRv.adapter = adapter
+        binding.cinemaRv.adapter = pagingAdapter
+
+        initListeners()
 
-        viewModel.moviesList.observe(viewLifecycleOwner, ::setResult)
     }
 
-    private fun setResult(result: DataResult<List<Movie>>) =
-        when (result) {
-            is DataResult.Success<List<Movie>> -> {
-                adapter.submitList(result.value)
-            }
-            is DataResult.EmptyResult -> {
-                showShortToast(R.string.empty_movies_list)
-            }
-            is DataResult.Error -> {
-                Log.e("FragmentMoviesList", "getMoviesList: Failed", result.error)
-                showShortToast(R.string.something_wrong)
+    private fun initListeners() {
+
+        binding.textInputLayout.editText?.doOnTextChanged { inputText, _, _, _ ->
+            viewModel.loadData(inputText.toString())
+            pagingAdapter.refresh()
```

Comment:
pagingAdapter.refresh() should be done just after debounce and if something was changed in input